### PR TITLE
[Merged by Bors] - feat: add notation for Real.sqrt

### PR DESCRIPTION
**Pull request by eric-wieser**
This adds the notation `√r` for `Real.sqrt r`. The precedence is such that `√x⁻¹` is parsed as `√(x⁻¹)`; not because this is particularly desirable, but because it's the default and the choice doesn't really matter.

This is extracted from #7907, which adds a more general nth root typeclass.
The idea is to perform all the boring substitutions downstream quickly, so that we can play around with custom elaborators with a much slower rate of code-rot.
This PR also won't rot as quickly, as it does not forbid writing `x.sqrt` as that PR does.

While perhaps claiming `√` for `Real.sqrt` is greedy; it:
* Is far more common thatn `NNReal.sqrt` and `Nat.sqrt`
* Is far more interesting to mathlib than `sqrt` on `Float`
* Can be overloaded anyway, so this does not prevent downstream code using the notation on their own types.
* Will be replaced by a more general typeclass in a future PR.

[Zulip](https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/.60Sqrt.60.20notation.20typeclass/near/400086502)

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by urkud** [comment]
LGTM. Should we wait for another maintainer, or just merge it?

**Review by Ruben-VandeVelde — APPROVED** [review]
Pretty.

maintainer merge

**Review by j-loreaux — COMMENTED** [review]
To be honest, it's cute, but I'm not sure it improves readability in large expressions. 

bors merge

**Comment by j-loreaux** [comment]
bors d+

**Comment by urkud** [comment]
@eric-wieser Should we merge it now or you plan to do more changes?

**Comment by eric-wieser** [comment]
@fpvandoorn expressed some concerns about notation scope on Zulip, but I think they might need revisiting with your notation class PR anyway; so I'll just put this in. If the notation starts causing conflicts, we can revisit the notation being global.

bors merge

**Pull request closed without merging**
